README.md

```diff
@@ -0,0 +1,74 @@
+# rust-kzg-bn254
+
+## Description
+
+This library offers a set of functions for generating and interacting with bn254 KZG commitments and proofs in rust, with the motivation of supporting fraud and validity proof logic in EigenDA rollup integrations.
+
+## Crates
+
+This repository is organized as a Rust workspace with three main crates:
+
+### [rust-kzg-bn254-primitives](./primitives)
+
+[![Docs](https://docs.rs/rust-kzg-bn254-primitives/badge.svg)](https://docs.rs/rust-kzg-bn254-primitives/latest/rust_kzg_bn254_primitives/)
+[![Crate](https://img.shields.io/crates/v/rust-kzg-bn254-primitives.svg)](https://crates.io/crates/rust-kzg-bn254-primitives)
+
+Provides the fundamental data structures and operations:
+- `Blob`: Data representation with conversion methods
+- `Polynomial`: Support for both evaluation and coefficient forms
+- Various arithmetic and helper functions
+
+### [rust-kzg-bn254-prover](./prover)
+
+[![Docs](https://docs.rs/rust-kzg-bn254-prover/badge.svg)](https://docs.rs/rust-kzg-bn254-prover/latest/rust_kzg_bn254_prover/)
+[![Crate](https://img.shields.io/crates/v/rust-kzg-bn254-prover.svg)](https://crates.io/crates/rust-kzg-bn254-prover)
+
+Implements KZG commitment and proof generation:
+- `KZG`: Main struct for creating commitments and generating proofs
+- `SRS`: Structured Reference String handling
+- Optimized parallel FFT implementations
+
+### [rust-kzg-bn254-verifier](./verifier)
+
+[![Docs](https://docs.rs/rust-kzg-bn254-verifier/badge.svg)](https://docs.rs/rust-kzg-bn254-verifier/latest/rust_kzg_bn254_verifier/)
+[![Crate](https://img.shields.io/crates/v/rust-kzg-bn254-verifier.svg)](https://crates.io/crates/rust-kzg-bn254-verifier)
+
+Provides verification functions:
+- Single proof verification
+- Batch verification for improved efficiency
+
+## Getting Started
+
+For a complete end-to-end example, see the `test_compute_kzg_proof` function in [prover/tests/kzg_test.rs](./prover/tests/kzg_test.rs).
+
+### Building and Benchmark
+
+```bash
+# Build all crates
+cargo build
+
+# Run benchmarks
+cargo bench
+```
+### Compatibility
+1. The project is compatible with Rust 1.75 or later
+
+### Downstream Dependencies
+1. Arbitrum Nitro uses rust 1.78
+2. RiscZero ZKVM uses 1.85
+3. SP1 ZKVM uses 1.79
+
+
+### Notes on testing
+If you encounter issue running tests and it failes with the following error:
+```
+error: package `half v2.5.0` cannot be built because it requires rustc 1.81 or newer, while the currently active rustc version is 1.75.0
+```
+then run `cargo update half@2.5.0 --precise 2.4.1` to update this transitive dependency to a lower version.
```

Comment:
```suggestion
then run `cargo update half --precise 2.4.1` to downgrade this transitive dependency to a lower version that works with our MSRV.
```

Reply:
no need to pin to @2.5.0 because likely 2.5.1 or higher versions will be out soon, making this command outdated.